src/lovok_handle_internal.cpp

```diff
@@ -67,8 +67,14 @@ bool SubtractUnderflow(uint64_t a, uint64_t b, uint64_t *result) {
     return true;
 }
 
-LovokStatusCode ParseBoxes(FileWrapper *fileWrapper, uint64_t length, uint64_t byteOffset, const std::function<LovokStatusCode(const Box &, uint64_t)> &f) {
-    if (length < 0) { return PARSE_ERROR; }
+LovokStatusCode ParseBoxes(FileWrapper *fileWrapper, uint64_t length, uint64_t byteOffset, uint64_t headerSize, const std::function<LovokStatusCode(const Box &, uint64_t)> &f) {
+    if (!AddOverflow(byteOffset, headerSize, &byteOffset)) {
+        return INVALID_FILE;
+    }
+
+    if (!SubtractUnderflow(length, headerSize, &length)) {
+        return INVALID_FILE;
+    }
     while (length > 0) {
         Box header = Box();
         LovokStatusCode parsedHeader = ParseHeader(fileWrapper, &header);
```

Comment:
I think I opened a separate issue because I think calling ParseHeader without some additional size checks doesn't seem safe, seems like a read flag.

Reply:
I agree, I will take it up in a separate PR.